carrental/views.py

```diff
@@ -0,0 +1,295 @@
+from django.http.response import HttpResponseRedirect
+from django.shortcuts import render, redirect
+from django.contrib.auth.models import User
+from .models import *
+from django.contrib.auth import authenticate, login, logout
+from django.contrib.auth.decorators import login_required
+
+
+def index(request):
+    cars = Car.objects.all()
+    return render(request, "index.html", {'cars': cars})
+
+
+def customer_signup(request):
+    if request.user.is_authenticated:
+        return redirect("/")
+    if request.method == "POST":
+        username = request.POST['username']
+        email = request.POST['email']
+        first_name = request.POST['first_name']
+        last_name = request.POST['last_name']
+        password1 = request.POST['password1']
+        password2 = request.POST['password2']
+        phone = request.POST['phone']
+        city = request.POST['city']
+
+        if password1 != password2:
+            return redirect("/customer_signup")
+
+        user = User.objects.create_user(username=username, email=email, first_name=first_name, last_name=last_name,
+                                        password=password1)
+        user.save()
+        try:
+            location = Location.objects.get(city=city.lower())
+        except:
+            location = None
+        if location is not None:
+            customer = Customer(user=user, phone=phone, location=location, type="Customer")
+        else:
+            location = Location(city=city.lower())
+            location.save()
+            location = Location.objects.get(city=city.lower())
+            customer = Customer(user=user, phone=phone, location=location, type="Customer")
+        customer.save()
+        alert = True
+        return render(request, "customer_signup.html", {'alert': alert})
+    return render(request, "customer_signup.html")
+
+
+def customer_login(request):
+    if request.user.is_authenticated:
+        return redirect("/")
+    else:
+        if request.method == "POST":
+            username = request.POST['username']
+            password = request.POST['password']
+            user = authenticate(username=username, password=password)
+
+            if user is not None:
+                user1 = Customer.objects.get(user=user)
+                if user1.type == "Customer":
+                    login(request, user)
+                    return redirect("/customer_homepage")
```

Comment:
to jest dziwne, skoro dostajesz usera z `authenticate` to po co go potem pobierasz przez `get`, możesz się dobrać przez połączenie, a dwa to po co ta klasa `Customer` wydaje się nadmiarowa, a trzy to chcemy hardkodować takich rzeczy jak ten `Customer`, już abstrahując, że skoro masz klasę `Customer` to jak miałby to nie być `Customer`? To jednak chcemy mieć takie coś zdefiniowane, a potem się tylko do tego odwołujemy, nigdy nie hardkodujemy